sbt

```diff
@@ -148,7 +148,9 @@ make_url() {
 make_coursier_url () {
   local version="$1"
 
-  echo "https://github.com/coursier/sbt-launcher/releases/download/v$version/csbt"
+  local base="${sbt_launch_repo:-$(url_base "$version")}"
+
+  echo "$base/io/get-coursier/sbt-launcher_2.12/$version/sbt-launcher_2.12-$version.jar"
```

Comment:
There's no `io/get-coursier/sbt-launcher_2.12` anything under https://repo.scala-sbt.org/scalasbt/maven-releases.

And even if this were point to https://repo1.maven.org/maven2/io/get-coursier/sbt-launcher_2.12/1.2.19/sbt-launcher_2.12-1.2.19.jar I'm not 100% sure if that's equivalent to https://github.com/coursier/sbt-launcher/releases/download/v1.2.19/csbt (I remember something about Alex saying that there's a specific process in packaging coursier/sbt-launcher.

Reply:
@alexarchambault so what needs to be done to make `coursier/sbt-launcher` equivalent to `csbt`?

Reply:
> And even if this were point to https://repo1.maven.org/maven2/io/get-coursier/sbt-launcher_2.12/1.2.19/sbt-launcher_2.12-1.2.19.jar I'm not 100% sure if that's equivalent to https://github.com/coursier/sbt-launcher/releases/download/v1.2.19/csbt (I remember something about Alex saying that there's a specific process in packaging coursier/sbt-launcher.

Yes, the `csbt` jar at https://github.com/coursier/sbt-launcher/releases/download/v1.2.19/csbt is generated roughly with
```bash
$ cs bootstrap -o csbt sbt-launcher
```
It's a coursier ["bootstrap"](https://get-coursier.io/docs/cli-bootstrap.html#bootstraps), containing a list of URLs, which includes https://repo1.maven.org/maven2/io/get-coursier/sbt-launcher_2.12/1.2.19/sbt-launcher_2.12-1.2.19.jar and its dependencies.

Reply:
@sideeffffect IIUC, you'd like this JAR (csbt) to be downloaded from a repository of yours, rather than from GitHub?

For that to work, I think you'll have to generate your own `csbt`, like
```bash
$ cs bootstrap sbt-launcher:1.2.19 -o csbt --no-default -r https://your-custom-repo -r htttps://other-custom-repo
```

I guess that scenario would be made easier if / when the official sbt launcher [were to be reworked](https://github.com/sbt/sbt/issues/5406) based on https://github.com/coursier/sbt-launcher.